service/payment.go

```diff
@@ -100,7 +111,69 @@ func (service *PaymentService) CreatePaymentSession(w http.ResponseWriter, req *
 	// log.Trace("TODO log successful creation with details") // TODO
 }
 
-func getPaymentResource(w http.ResponseWriter, req *http.Request, resource string, cfg *config.Config) (*models.PaymentResource, error) {
+// GetPaymentSession retrieves the payment session
+func (service *PaymentService) GetPaymentSession(w http.ResponseWriter, req *http.Request) {
+	id := req.URL.Query().Get(":payment_id")
+	if id == "" {
+		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	paymentResource, err := service.DAO.GetPaymentResource(id)
+	if err != nil {
+		if err == mgo.ErrNotFound {
```

Comment:
Should the service 'know' about Mongo? This logic could be moved to the DAO, and a `nil` paymentResource returned to the service.